### PR TITLE
[dhcp-relay] Add support for dhcp_relay config cli

### PR DESCRIPTION
**Pull request by yaqiangz**
#### Why I did it
Currently the config cli of dhcpv4 is may cause confusion and config of dhcpv6 is missing.

#### How I did it
Add dhcp_relay config cli and test cases.
```
config dhcp_relay ipv4 helper (add | del) <vlan_id> <helper_ip_list>
config dhcp_relay ipv6 destination (add | del) <vlan_id> <destination_ip_list>
```
3. Updated docs for it in sonic-utilities: https://github.com/sonic-net/sonic-utilities/pull/2598/files

#### How to verify it
Build docker-dhcp-relay.gz with and without INCLUDE_DHCP_RELAY, and check target/docker-dhcp-relay.gz.log
```
[ REASON ] :      target/docker-dhcp-relay.gz does not exist   NON-EXISTENT PREREQUISITES: docker-start target/docker-config-engine-bullseye.gz-load target/python-wheels/bullseye/sonic_utilities-1.2-py3-none-any.whl-install target/debs/bullseye/python3-swsscommon_1.0.0_amd64.deb-install 
[ FLAGS  FILE    ] : [] 
[ FLAGS  DEPENDS ] : [] 
[ FLAGS  DIFF    ] : [] 
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /sonic/dockers/docker-dhcp-relay/cli-plugin-tests, configfile: pytest.ini
plugins: pyfakefs-5.0.0, cov-2.10.1
collecting ... collected 35 items

test_config_dhcp_relay.py::TestConfigDhcpRelay::test_plugin_registration PASSED [  2%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_test_config_dhcp_relay_add_del_with_nonexist_vlanid[ipv4-add] PASSED [  5%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_dhcp_relay_with_invalid_ip[ipv4-add] PASSED [  8%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_duplicate_dhcp_relay[ipv4-add] PASSED [ 11%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_test_config_dhcp_relay_add_del_with_nonexist_vlanid[ipv4-del] PASSED [ 14%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_dhcp_relay_with_invalid_ip[ipv4-del] PASSED [ 17%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_duplicate_dhcp_relay[ipv4-del] PASSED [ 20%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_test_config_dhcp_relay_add_del_with_nonexist_vlanid[ipv6-del] PASSED [ 22%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_dhcp_relay_with_invalid_ip[ipv6-del] PASSED [ 25%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_duplicate_dhcp_relay[ipv6-del] PASSED [ 28%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_test_config_dhcp_relay_add_del_with_nonexist_vlanid[ipv6-add] PASSED [ 31%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_dhcp_relay_with_invalid_ip[ipv6-add] PASSED [ 34%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_duplicate_dhcp_relay[ipv6-add] PASSED [ 37%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_dhcp_with_exist_ip[ipv6] PASSED [ 40%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_del_nonexist_dhcp_relay[ipv6] PASSED [ 42%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_dhcp_relay[ipv6] PASSED [ 45%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_multiple_dhcp_relay[ipv6] PASSED [ 48%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_dhcp_with_exist_ip[ipv4] PASSED [ 51%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_del_nonexist_dhcp_relay[ipv4] PASSED [ 54%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_dhcp_relay[ipv4] PASSED [ 57%]
test_config_dhcp_relay.py::TestConfigDhcpRelay::test_config_add_del_multiple_dhcp_relay[ipv4] PASSED [ 60%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_plugin_registration PASSED [ 62%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_dhcp_relay_with_nonexist_vlanid PASSED [ 65%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_dhcp_relay_with_invalid_vlanid PASSED [ 68%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_dhcp_relay_with_invalid_ip PASSED [ 71%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_dhcp_relay_with_exist_ip PASSED [ 74%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_del_dhcp_relay_dest PASSED [ 77%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_del_dhcpv6_relay_dest PASSED [ 80%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_add_del_multiple_dhcpv6_relay_dest PASSED [ 82%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_remove_nonexist_dhcp_relay_dest PASSED [ 85%]
test_config_vlan_dhcp_relay.py::TestConfigVlanDhcpRelay::test_config_vlan_remove_dhcp_relay_dest_with_nonexist_vlanid PASSED [ 88%]
test_show_dhcp6relay_counters.py::TestDhcp6RelayCounters::test_show_counts PASSED [ 91%]
test_show_dhcp_relay.py::TestVlanDhcpRelay::test_plugin_registration PASSED [ 94%]
test_show_dhcp_relay.py::TestVlanDhcpRelay::test_dhcp_relay_column_output PASSED [ 97%]
test_show_dhcpv6_helper.py::TestDhcpRelayHelper::test_show_dhcpv6_helper_0_DHCPv6_Helpers <- ../../../../usr/local/lib/python3.9/dist-packages/pyfakefs/fake_filesystem_unittest.py PASSED [100%]

=============================== warnings summary ===============================
/usr/lib/python3/dist-packages/_pytest/junitxml.py:446
  /usr/lib/python3/dist-packages/_pytest/junitxml.py:446: PytestDeprecationWarning: The 'junit_family' default value will change to 'xunit2' in pytest 6.0. See:
    https://docs.pytest.org/en/stable/deprecations.html#junit-family-default-value-change-to-xunit2
  for more information.
    _issue_warning_captured(deprecated.JUNIT_XML_DEFAULT_FAMILY, config.hook, 2)

/usr/local/lib/python3.9/dist-packages/tabulate.py:7
  /usr/local/lib/python3.9/dist-packages/tabulate.py:7: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import namedtuple, Iterable

-- Docs: https://docs.pytest.org/en/stable/warnings.html
- generated xml file: /sonic/dockers/docker-dhcp-relay/cli-plugin-tests/test-results.xml -

----------- coverage: platform linux, python 3.9.2-final-0 -----------
Name                                                                                        Stmts   Miss  Cover
---------------------------------------------------------------------------------------------------------------
conftest.py                                                                                    15      0   100%
test_config_dhcp_relay.py                                                                     129      0   100%
test_config_vlan_dhcp_relay.py                                                                151      0   100%
/sonic/dockers/docker-dhcp-relay/cli/config/plugins/dhcp_relay.py                             191     21    89%

======================= 35 passed, 2 warnings in 20.49s ========================
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Comment by yaqiangz** [comment]
Hi @kellyyeh @jcaiMR, could you help to review this PR?

File: dockers/docker-dhcp-relay/cli/config/plugins/vlan_dhcp_relay.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    if len(dhcp_servers):
+        vlan['dhcp_servers'] = dhcp_servers
+    if len(dhcpv6_servers):
+        vlan['dhcpv6_servers'] = dhcpv6_servers
+
+    db.cfgdb.set_entry('VLAN', vlan_name, vlan)
+
+    if len(added_servers):
+        click.echo("Added DHCP relay destination addresses {} to {}".format(added_servers, vlan_name))
+        try:
```
Comment (Blueve) [inline dockers/docker-dhcp-relay/cli/config/plugins/vlan_dhcp_relay.py:53]: Can we reuse this part? Why not use `restart_dhcp_relay_service`?

**Comment by Blueve** [comment]
Let's update cmd ref doc and link that PR to this one.
And update UT coverage rate for these 2 commands

File: dockers/docker-dhcp-relay/cli/config/plugins/vlan_dhcp_relay.py (hunk 1 above)
Comment (yaqiangz) [reply dockers/docker-dhcp-relay/cli/config/plugins/vlan_dhcp_relay.py:57]: Fixed

**Comment by yaqiangz** [comment]
Updated for cli docs: https://github.com/sonic-net/sonic-utilities/pull/2598/files

File: dockers/docker-dhcp-relay/cli-plugin-tests/test_config_dhcp_relay.py [hunk 2]
```diff
@@ -1,229 +1,265 @@
...
+
+
+class TestConfigDhcpRelay(object):
     def test_plugin_registration(self):
         cli = mock.MagicMock()
         dhcp_relay.register(cli)
-        cli.commands['vlan'].add_command.assert_called_once_with(dhcp_relay.vlan_dhcp_relay)
 
-    def test_config_vlan_add_dhcp_relay_with_nonexist_vlanid(self):
+    def test_test_config_dhcp_relay_add_del_with_nonexist_vlanid(self, ip_version, op):
```
Comment (kellyyeh) [inline dockers/docker-dhcp-relay/cli-plugin-tests/test_config_dhcp_relay.py:114]: Typo? test_test_config_dhcp_relay_add_del_with_nonexist_vlanid

Comment (yaqiangz) [reply dockers/docker-dhcp-relay/cli-plugin-tests/test_config_dhcp_relay.py]: Oh, it's a typo, I have fixed it.

**Review by kellyyeh — APPROVED** [review]

**Review by Blueve — APPROVED** [review]

**Comment by yaqiangz** [comment]
@yxieca Could you please help to review this PR?

**Review by yxieca — APPROVED** [review]

**Pull request merged**